src/compiler/program.ts

```diff
@@ -691,13 +735,23 @@ namespace ts {
         function getCommonSourceDirectory() {
             if (commonSourceDirectory === undefined) {
                 const emittedFiles = filter(files, file => sourceFileMayBeEmitted(file, options, isSourceFileFromExternalLibrary));
-                if (options.rootDir && checkSourceFilesBelongToPath(emittedFiles, options.rootDir)) {
-                    // If a rootDir is specified and is valid use it as the commonSourceDirectory
+                if (options.rootDir) {
+                    // If a rootDir is specified use it as the commonSourceDirectory
                     commonSourceDirectory = getNormalizedAbsolutePath(options.rootDir, currentDirectory);
                 }
+                else if (options.composite) {
+                    // Project compilations never infer their root from the input source paths
+                    Debug.assert(!!options.configFilePath);

```

Comment:
i would rather we do not assert here. if createProgram fails all LS operations will be broken from this point on. we can make this as a check, and fall back to the path of the first file for instance.. just not to throw.